neuralpp/test/quick_tests/graphical_model/representation/factor/symbolic_factor_test.py

```diff
@@ -54,8 +104,26 @@ def test_sum_out_variable():
 
     symbolic = SymbolicFactor([x, y], expression1)
     sum_out_x = symbolic.sum_out_variable(x)
+
     assert sum_out_x.expression.sympy_object == 3 * y_symbol
 
+def test_if_then_else_sum_out_variable():
+    x = IntegerVariable("x", 3)
+    y = IntegerVariable("y", 2)
+
+    x_symbol, y_symbol = sympy.symbols("x y")
+    x_sympy = SymPyVariable(x_symbol, int)
+    y_sympy = SymPyVariable(y_symbol, int)
+
+    expression1 = if_then_else(x_sympy > y_sympy, 2, 3)
+
+    symbolic = SymbolicFactor([x, y], expression1)
+    sum_out_x = symbolic.sum_out_variable(x)
```

Comment:
Suggestion: you can use `^` for `sum_out_variable`: `symbolic ^ x`. It also works if the second argument is an `Iterable` of variables.